File: Kotlin/FizzBuzz_v1.kt
// KotlinでFizzBuzz
// kotlinc hello.kt -include-runtime -d hello.jar コンパイル
// java -jar hello.jar 実行
// kotlinc-js hello.kt -output hello.js JavaScriptにコンパイルも可

// var 変数名   ミュータブル
// val 変数名   イミュータブル

fun FizzBuzz() {
    for (i in 0..100){
        if (i % 3 == 0 && i % 5 == 0){
            println("${i} FizzBuzz.")
        }else if (i % 3 == 0){
            println("${i} Fizz.")
        }else if (i % 5 == 0){
            println("${i} Buzz.")
        }else{
            println(i)
        }
    }
}

fun main() {
    FizzBuzz()
}

File: Lua/FizzBuzz_v1.lua
for i = 0, 100 do
    if i % 3 == 0 and i % 5 == 0 then
        data = string.format("%d FizzBuzz.",i)
        print(data)
    elseif i % 3 == 0 then
        data = string.format("%d Fizz.",i)
        print(data)
    elseif i % 5 == 0 then
        data = string.format("%d Buzz.",i)
        print(data)
    else
        print(i)
    end
end
